### PR TITLE
[Rule Tuning] M365/Azure Brute-Forcing New Rule and Tuning; Deprecate Similar Rule

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*: 
* https://github.com/elastic/ia-trade-team/issues/271
* https://github.com/elastic/ia-trade-team/issues/272

## Summary - What I changed
The following changes have been made as a result of community slack [thread](https://elasticstack.slack.com/archives/C016E72DWDS/p1725544493561649).

* Stage 1 deprecation `Potential Password Spraying of Microsoft 365 User Accounts`: This rule's query logic overlaps with `Attempts to Brute Force a Microsoft 365 User Account`, however is a threshold rule. With ES|QL we can uses the `STATS` command and `COUNT()` function to accomplish the same anomaly detection.
    * Once deprecation prefix is added and rule update is released, we will move the rule to deprecation folder 
* `Attempts to Brute Force a Microsoft 365 User Account` has been tuned to count the number of sources for each failed login. 
* Added another rule that focuses on Azure data source, specifically Entra ID sign-in logs. - `Azure Entra Sign-in Brute Force against Microsoft 365 Accounts`

Note: For the Azure rule, we should update the telemetry filterlist and monitor `azure.signinlogs.properties.status.error_code` for tuning. This will be the most concrete way to remove false-positives and focus solely on brute force related [error codes](https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes) that would generate, however, at this time there are too many plausible error codes to scope for this PR.

## How To Test
Screenshot 2024-09-06 at 12 00 52 PM (2)

Screenshot 2024-09-06 at 12 00 57 PM

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [x] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: Tuning - Guidelines

These guidelines serve as a reminder set of considerations when tuning an existing rule.

### Documentation and Context

- [ ] Detailed description of the suggested changes.
- [ ] Provide example JSON data or screenshots.
- [ ] Provide evidence of reducing benign events mistakenly identified as threats (False Positives).
- [ ] Provide evidence of enhancing detection of true threats that were previously missed (False Negatives).
- [ ] Provide evidence of optimizing resource consumption and execution time of detection rules (Performance).
- [ ] Provide evidence of specific environment factors influencing customized rule tuning (Contextual Tuning).
- [ ] Provide evidence of improvements made by modifying sensitivity by changing alert triggering thresholds (Threshold Adjustments).
- [ ] Provide evidence of refining rules to better detect deviations from typical behavior (Behavioral Tuning).
- [ ] Provide evidence of improvements of adjusting rules based on time-based patterns (Temporal Tuning).
- [ ] Provide reasoning of adjusting priority or severity levels of alerts (Severity Tuning).
- [ ] Provide evidence of improving quality integrity of our data used by detection rules (Data Quality).
- [ ] Ensure the tuning includes necessary updates to the release documentation and versioning.

### Rule Metadata Checks

- [ ] `updated_date` matches the date of tuning PR merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive. Review to ensure the original intent of the rule is maintained.

### Testing and Validation

- [ ] Validate that the tuned rule's performance is satisfactory and does not negatively impact the stack.
- [ ] Ensure that the tuned rule has a low false positive rate.

File: rules/integrations/azure/credential_access_entra_signin_brute_force_microsoft_365.toml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  and event.category == "authentication"
+  and to_lower(azure.signinlogs.properties.resource_display_name) rlike "(.*)365(.*)"
+  and azure.signinlogs.category in ("NonInteractiveUserSignInLogs", "SignInLogs")
+  and event.outcome != "success"
+  // for tuning review azure.signinlogs.properties.status.error_code
+  // https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes
+// count the number of login sources and failed login attempts
+| stats
+  login_source_count = count(source.ip),
+  failed_login_count = count(*) by target_time_window, azure.signinlogs.properties.user_principal_name
```
Comment (Samirbous) [inline rules/integrations/azure/credential_access_entra_signin_brute_force_microsoft_365.toml:63]: this rule covers the bruteforce targeting one user account, what about a new rule same source.ip targeting many user accounts (join by source.ip only or ASN)

**Review by Samirbous — APPROVED** [review]

File: rules/integrations/azure/credential_access_entra_signin_brute_force_microsoft_365.toml (hunk 1 above)
Comment (terrancedejesus) [reply rules/integrations/azure/credential_access_entra_signin_brute_force_microsoft_365.toml:63]: Yea for sure I'll add one for that as well!

File: rules/integrations/o365/credential_access_microsoft_365_potential_password_spraying_attack.toml [hunk 2]
```diff
@@ -21,10 +21,8 @@ from = "now-30m"
 index = ["filebeat-*", "logs-o365*"]
 language = "kuery"
 license = "Elastic License v2"
-name = "Potential Password Spraying of Microsoft 365 User Accounts"
-note = """## Setup
-
-The Office 365 Logs Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+name = "Deprecated - Potential Password Spraying of Microsoft 365 User Accounts"
```
Comment (Mikaayenson) [inline rules/integrations/o365/credential_access_microsoft_365_potential_password_spraying_attack.toml:24]: 👍

**Review by Mikaayenson — APPROVED** [review]
lgtm. any thought on a setup guide?

**Comment by terrancedejesus** [comment]
> lgtm. any thought on a setup guide?

@Mikaayenson great question and thanks for taking a look. Both rules link to the Azure integration, which has it's own setup and configuration documentation. Ref - https://www.elastic.co/docs/current/integrations/azure

There are no special configurations that we did to gain visibility into these actions as they are the default Entra ID logs that stream to the Azure event hub. If we want to replicate steps from the integration into rules, my questions would be:

1. Should we only add these if the rule required a more custom setup or maybe specific integration advanced settings that are not enabled by default?
2. If we replicate instructions from integration docs, they may become outdated and desync from integration docs, thus should we rely on security docs?

Happy to do either, but for these specifically I am in favor of leaning on the Azure integration docs as we do no special configuration setup outside of that.

**Comment by Mikaayenson** [comment]
> e want to replicate steps from the integration into rules, my questions would be:
> 
> 1.

Makes sense. I agree. We can leave as-is then. Im also wondering if we should add the link to the references or something, but maybe its already assumed that you should have followed those steps before using the rule.

**Pull request merged**